sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -368,7 +384,17 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
     this.topicResourceSerializer = new TopicResourceSerializer();
     this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
     this.ruleResourceSerializer = new RuleResourceSerializer();
-    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") ||
+      connectionString.match("Endpoint=http://(.*)/;") ||
+      connectionString.match("Endpoint=https://(.*)/;") ||
+      "")[1];
```

Comment:
What is the reason to support http and https in the user provided connection string?
From what we know, the connection string is picked up from the portal and it always has the `sb://` scheme

Reply:
I recall coming across some usage examples where the connection string was using a different protocol. 
Doesn't seem reasonable to make hard assumptions on where user gets the connection string from?

Reply:
Please provide pointers to usage examples where the connection string for Service Bus used a different protocol

Reply:
Unable to find them now, it was on a blog post for some user example I think..
I updated it on local and was able to verify that any other protocol would work as well. (I.e., almost all tests pass when run for 10-15 min). 
This is because we mainly work with the host name.
(Although, in amqp-common too we hardcode to check only the sb:// prefix. 
It currently doesn't seem to be an issue as Event Hubs on portal uses sb:// prefix as well. But we might want to handle multiple protocols.)

Is there any documentation that says the endpoint must only use the sb:// protocol?
It seems like the endpoint could be any valid FQDN. Since storage uses http:// prefix ones in the endpoint as well, and so some automated, complex solution handling all endpoints could end up with using this.

Also, both our SDK (amqp-common) and .NET SDK normalizes the endpoint from connection string to have a trailing slash, IF it doesn't have it - this supports theory that the Endpoint could be constructed manually as well.

Reply:
Also carried out some more testing on local, the URIs can be FQDNs using http:// https:// and sb:// protocols.
Using an invalid protocol prefix results in the forward related feature throw an error. So it seems like we should be supporting other protocols and not mislead into having to use sb:// only

If users wanted SSL authentication, would that be a case where they specifically end up wanting to use https:// in the endpoint?

I'm okay to revert changes if we want to track the protocols support in a different issue maybe?

.NET SDK's connection string parser seems to factor in `amqps://` protocol in addition.

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/tests/ServiceBusConnectionStringBuilderTests.cs

We might want to revisit this to see if we are indeed handling the cornercases and whitespaces etc.

cc: @AlexGhiondea

Reply:
> Using an invalid protocol prefix results in the forward related feature throw an error. So it seems like we should be supporting other protocols and not mislead into having to use sb:// only

Using http, https, sb or amqps when passing absolute urls to the forward related properties would still work without us trying to explicitly support more than `sb://` in the connection string. This is due to us not mutating user given input when it is an absolute url.

If we want to explore supporting more than just `sb://` in the connection string, I would suggest that be done as part of a separate work item. Having `ServiceBusManagementClient` support all 3 schemes, but `ServiceusClient` support only `sb://` is not a state we should be in.